{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca5139d1-63f2-46f9-a2e7-c05206995cbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "import posixpath\n",
    "from pygenomeviz import GenomeViz\n",
    "from pygenomeviz.parser import Gff\n",
    "from pygenomeviz.parser import Genbank\n",
    "from pygenomeviz.align import MMseqs\n",
    "from pygenomeviz.utils import load_example_gff_file\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08306a6d-f2eb-4156-a09b-26ab8ad47d79",
   "metadata": {},
   "outputs": [],
   "source": [
    "gbk_file = [\"./frag_plasmids_N25_2.gbff\",\"./frag_plasmids_R848_2.gbff\",\"./frag_plasmids_Q97_2.gbff\",\"./NDM_plasmid.gbff\"]\n",
    "gbk_path = [Path(path) for path in gbk_file]\n",
    "gbk_list = list(map(Genbank, gbk_path))\n",
    "\n",
    "gv = GenomeViz(fig_track_height=1.2, feature_track_ratio=0.5, fig_width=70)\n",
    "gv.set_scale_xticks(labelsize=50)\n",
    "\n",
    "# Plot CDS features\n",
    "for gbk in gbk_list:\n",
    "    track = gv.add_feature_track(gbk.name, gbk.get_seqid2size(), align_label=False,labelsize=50)\n",
    "    for seqid, features in gbk.get_seqid2features(\"CDS\").items():\n",
    "        segment = track.get_segment(seqid)\n",
    "        for feature in features:\n",
    "            gene_name = str(feature.qualifiers.get(\"gene\", [\"\"])[0])\n",
    "    # Set user-defined feature color based on gene name\n",
    "            if gene_name.startswith(\"iuc\"):\n",
    "                color = \"#74C376\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in (\"iutA\", \"rmpA\"):\n",
    "                color = \"#74C376\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in (\"msr(E)\", \"mph(E)\", \"armA\", \"mph(A)\", \"sul1\", \"sul2\",\"aph(3')-la\",\"aph(3')-VI\",\"dfrA5\"):\n",
    "                color = \"#9A1E13\"\n",
    "                segment.add_features(feature, plotstyle=\"bigarrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in ( \"yqaJ\",\"mobI\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"tnp\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"tra\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            else:\n",
    "                color = \"#6A51A3\"\n",
    "                segment.add_features(feature, plotstyle=\"box\", color=color)\n",
    "\n",
    "# Run MMseqs RBH search\n",
    "align_coords = MMseqs(gbk_list).run()\n",
    "\n",
    "# Plot MMseqs RBH search links\n",
    "if len(align_coords) > 0:\n",
    "    min_ident = int(min([ac.identity for ac in align_coords if ac.identity]))\n",
    "    color, inverted_color = \"#0070FF\", \"red\"\n",
    "    for ac in align_coords:\n",
    "        gv.add_link(ac.query_link, ac.ref_link, color=color, inverted_color=inverted_color, v=ac.identity, vmin=min_ident, curve=True)\n",
    "    gv.set_colorbar([color, inverted_color], vmin=min_ident)\n",
    "\n",
    "fig = gv.plotfig()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f6f033c-7d1e-49c6-8aa1-e6510c3ca1ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "gv.savefig(savefile=\"./kp_mosaic_plasmid.jpg\",dpi=300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2ff0e97-89f8-4e1a-abdb-bccbf93ff41d",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89baa5e8-a8f5-4cd7-a2ef-c18938e01244",
   "metadata": {},
   "outputs": [],
   "source": [
    "gbk_file = [\"./frag_plasmids_JZ68_6.gbff\",\"./frag_plasmids_Q136_3.gbff\",\"./frag_plasmids_R374_3.gbff\",\"./OXA48_plasmid.gbff\"]\n",
    "gbk_path = [Path(path) for path in gbk_file]\n",
    "gbk_list = list(map(Genbank, gbk_path))\n",
    "\n",
    "gv = GenomeViz(fig_track_height=1.2, feature_track_ratio=0.5, fig_width=70)\n",
    "gv.set_scale_xticks(labelsize=50)\n",
    "\n",
    "# Plot CDS features\n",
    "for gbk in gbk_list:\n",
    "    track = gv.add_feature_track(gbk.name, gbk.get_seqid2size(), align_label=False,labelsize=50)\n",
    "    for seqid, features in gbk.get_seqid2features(\"CDS\").items():\n",
    "        segment = track.get_segment(seqid)\n",
    "        for feature in features:\n",
    "            gene_name = str(feature.qualifiers.get(\"gene\", [\"\"])[0])\n",
    "    # Set user-defined feature color based on gene name\n",
    "            if gene_name.startswith(\"iuc\"):\n",
    "                color = \"#74C376\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in (\"iutA\", \"rmpA\"):\n",
    "                color = \"#74C376\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in (\"msr(E)\", \"mph(E)\", \"armA\", \"mph(A)\", \"sul1\", \"sul2\",\"aph(3')-la\",\"aph(3')-VI\",\"dfrA5\"):\n",
    "                color = \"#9A1E13\"\n",
    "                segment.add_features(feature, plotstyle=\"bigarrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name in ( \"yqaJ\",\"mobI\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"aph\"):\n",
    "                color = \"#9A1E13\"\n",
    "                segment.add_features(feature, plotstyle=\"bigarrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"bla\"):\n",
    "                color = \"#9A1E13\"\n",
    "                segment.add_features(feature, plotstyle=\"bigarrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"tnp\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            elif gene_name.startswith(\"tra\"):\n",
    "                color = \"#4090C0\"\n",
    "                segment.add_features(feature, plotstyle=\"arrow\", color=color, label_type=\"gene\",text_kws=dict(size=25, color=color))\n",
    "            else:\n",
    "                color = \"#6A51A3\"\n",
    "                segment.add_features(feature, plotstyle=\"box\", color=color)\n",
    "\n",
    "# Run MMseqs RBH search\n",
    "align_coords = MMseqs(gbk_list).run()\n",
    "\n",
    "# Plot MMseqs RBH search links\n",
    "if len(align_coords) > 0:\n",
    "    min_ident = int(min([ac.identity for ac in align_coords if ac.identity]))\n",
    "    color, inverted_color = \"#0070FF\", \"red\"\n",
    "    for ac in align_coords:\n",
    "        gv.add_link(ac.query_link, ac.ref_link, color=color, inverted_color=inverted_color, v=ac.identity, vmin=min_ident, curve=True)\n",
    "    gv.set_colorbar([color, inverted_color], vmin=min_ident)\n",
    "\n",
    "fig = gv.plotfig()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf614a98-a552-4487-b6eb-2ad97e036354",
   "metadata": {},
   "outputs": [],
   "source": [
    "gv.savefig(savefile=\"./kp_MDR_plasmid.jpg\",dpi=300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72856d3b-a11b-414b-b2d5-11fcca850cb5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
